### PR TITLE
refactored game method

### PR DESCRIPTION
**Pull request by galia1902**

### CODE REVIEW

File: app/controllers/games_controller.rb [hunk 1]
```diff
@@ -26,19 +26,14 @@ def game
...
     if @rounds[0].nil?
       @question = rand_quest
       @answers = Answer.where(question_id: @question.id).shuffle
       session[:tryout_answers] = @answers
-
     elsif @game.game_mode == "MCQ"
-      @question = rand_quest
-      @answers = Answer.where(question_id: @question.id).shuffle
-      @game = Game.find(@game.id)
+      redirect_to :action => 'mcq'
```
Comment (Grokepau) [inline app/controllers/games_controller.rb]: could u use the path (to make it more readable)

File: app/controllers/games_controller.rb [hunk 2]
```diff
@@ -47,7 +42,13 @@ def game
         @answers << Answer.new(answer_data)
       end
     end
+  end
+
 
+  def show_mcq_round
```
Comment (Grokepau) [inline app/controllers/games_controller.rb]: method must be created in the routes and must be the same path as redirect to in round !!

File: app/controllers/games_controller.rb [hunk 3]
```diff
@@ -47,7 +42,13 @@ def game
...
+    @question = rand_quest
+    @answers = Answer.where(question_id: @question.id).shuffle
+    @game = Game.find(@game.id)
```
Comment (Grokepau) [inline app/controllers/games_controller.rb]: i think you need to authorize game (not sure)

File: app/controllers/rounds_controller.rb [hunk 4]
```diff
@@ -11,6 +11,10 @@ def create
     authorize @round
     @round.save!
     redirect_to game_path(@game)
+    answer_position = session[:tryout_answers].find_index do |answer_data|
+      answer_data[:id] == round_params[:answer_id]
+    end
+    raise
```
Comment (Grokepau) [inline app/controllers/rounds_controller.rb]: please take raise out before merge

File: app/views/games/mcq.html.erb [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+        let newValue;
+        let color = an.style.backgroundColor;
+        let rgba = document.querySelector('.mcq-answer').style.backgroundColor.split(')');
+        let length = rgba.length - 1;
+        rgba[length] = '0.2)';
+        newValue = rgba.join();
+        an.style.border = `4px solid ${newValue}`
+      }
+
+      let handleChosen = (ans) => {
```
Comment (beingSCK) [inline app/views/games/mcq.html.erb]: need to handle issue where you click more than one button (in sequence)

**Comment by beingSCK** [comment]
@galia1902 - Also, do we need to keep this branch open? I get the sense that these changes are already part of one of your other branches (in the PR 'Implement choose only once per round').

**Pull request merged**